### ISSUE TITLE
In 2.x - is it possible to add metadata to a ServerReponse? Is it possible in a ServerInterceptor?

### ISSUE THREAD

**Issue by tweidema** [grpc/grpc-swift#2119]
### What are you trying to achieve?

I would like to add metadata to the response sent back to the client. Preferably in a ServerInterceptor.

### What have you tried so far?
Here is my attempt, but metadata is a get-only property
```
struct grpcServerTracingInterceptor: ServerInterceptor {
    let logger: Logger
    init(logger: Logger) {
        self.logger = logger
    }
    func intercept<Input: Sendable, Output: Sendable>(
        request: StreamingServerRequest<Input>,
        context: ServerContext,
        next: @Sendable (
           _ request: StreamingServerRequest<Input>,
           _ context: ServerContext
        ) async throws -> StreamingServerResponse<Output>
    ) async throws -> StreamingServerResponse<Output> {
        logger.info("Pre-intercept of \(context.descriptor.service):\(context.descriptor.method)")
        let callTime = Date()
        // Forward the request.
        var response = try await next(request, context)

        logger.info("Post-intercept of \(context.descriptor.service):\(context.descriptor.method)")
        let duration = callTime.timeIntervalSinceNow
        response.metadata.addString("gRPC;dur=\(duration)", forKey: "Server-Timing")
        return response
    }
}
```
The addString (2nd-last line) gives: Cannot use mutating member on immutable value: 'metadata' is a get-only property

**Comment by glbrntt** [comment]
`metadata` is a computed property on `StreamingServerResponse`, so to mutate it (at the moment) you need to go via `accepted` and mutate that. 

That'll solve your immediate problem, but I think we should add an API to make this more convenient.

**Comment by tweidema** [comment]
@glbrntt Thank you! Could you maybe help someone who does not yet have many months of Swift experience out with a few more details? Here is what I've tried. It compiles, but the client gets "the http/2 server reset the stream", so I'm obviously doing something wrong.
At the end of my ServerInterceptor (after the line "let duration = callTime.timeIntervalSinceNow" from above)
```
    let oldResult = response.accepted
    let old_res = try oldResult.get()
    var new_metadata = old_res.metadata
    new_metadata.addString("gRPC;dur=\(duration)", forKey: "Server-Timing")
    return StreamingServerResponse<Output>(metadata: new_metadata, producer: old_res.producer)
```

**Comment by glbrntt** [comment]
Absolutely :) 

Interesting, I _suspect_ that happens because `try oldResult.get()` throws. 

What you should do here is switch over `response.accepted`, modify the appropriate value for each case, and then assign that back to `response.accepted`:

```swift
var response = try await next(...)

switch response.accepted {
case .success(var contents):
  contents.metadata.addString(...)
  response.accepted = .success(contents)
case .failure(var error):
  error.metadata.addString(...)
  response.accepted = .failure(error)
}

return response
```

**Comment by glbrntt** [comment]
I just opened https://github.com/grpc/grpc-swift/pull/2120 to provide a convenience setter for this as well.

**Comment by tweidema** [comment]
Thank so much. Looks much better than my code :-) But I still get the error.
If I comment out the line
```
contents.metadata.addString(timingHeader, forKey: "Server-Timing")
```
the call completes normally. I tried adding a do/catch block around the switch, but vscode says the catch-block is unreachable because no errors are thrown in the do block.
I then tried shaving down both the header key and values and found the cause to any upper-case letters in the server-key.
so
```
contents.metadata.addString(timingHeader, forKey: "server-timing")
```
works, but
```
contents.metadata.addString(timingHeader, forKey: "Server-timing")
```
gives "the http/2 server reset the stream.
